### PR TITLE
Add Lagrange variant demo

### PR DESCRIPTION
**Pull request by mscroggs**
Added a Lagrange variants demo.

When built for the website the new demo looks something like this:
new-demo

### CODE REVIEW

**Comment by mscroggs** [comment]
This new demo uses matplotlib to draw its plots, In https://github.com/FEniCS/dolfinx/pull/2140/commits/144a1dc7f5edd8a60f043f4b243f09a1e68e9d49, I've used a try except to allow the demo to run if matplotlib is not installed. I'm not sure if this is the best thing to do or not. Opinions?

**Review by garth-wells — CHANGES_REQUESTED** [review]
Looks great. I pushed some text edits.

I'd prefer to not add images to the repository. Could the figures be generated when the documentation is produced?

File: python/demo/demo_lagrange_variants.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+from dolfinx import fem, mesh
+from ufl import ds, dx, grad, inner
+from dolfinx.fem import Function, FunctionSpace, assemble_scalar, form
+from dolfinx.mesh import create_unit_interval
+
+from mpi4py import MPI
+from petsc4py.PETSc import ScalarType
+
+try:
```
[the remark this answers is not in the document]
Comment (mscroggs) [reply python/demo/demo_lagrange_variants.py]: I've made the CIs install matplotlib in https://github.com/FEniCS/dolfinx/pull/2140/commits/7855b37fb935540e36b2aab2602d89708dd7ff71

**Comment by mscroggs** [comment]
> I'd prefer to not add images to the repository. Could the figures be generated when the documentation is produced?

I believe this is now done

**Review by garth-wells — APPROVED** [review]

**Pull request merged**